### ISSUE TITLE
Add `ChannelConfirmed` Event

### ISSUE THREAD

**Issue by TheBlueMatt** [lightningdevkit/rust-lightning#1394]
... (and maybe a `ChannelOpened/Accepted` event)?

**Comment by TheBlueMatt** [comment]
For `ChannelConfirmed` we should do it on funding-locked. For `ChannelOpened` (if we want one) we'd want to do it when the `AcceptChannel` message is either received or sent. The above schema looks good to me for the first, for the second we won't have a user id for inbounds, and the channel id will be the `temporary_channel_id` instead.

**Comment by wvanlint** [comment]
I have been taking a look at implementing `ChannelConfirmed`. From my understanding, the event should be emitted for incoming and outgoing channels when `channel_ready` messages have been both sent and received and the channel becomes usable. However, since it is possible to receive multiple `channel_ready` messages with different `alias` values, it's probably desirable to only emit the event when the channel state changes from non-usable to usable. To achieve that, it might be helpful to refactor the result of `Channel::channel_ready` but can be done without such a change as well. Does that sound correct?

**Comment by tnull** [comment]
@wvanlint Do you still intend to pick this up? Asking since I'd like to utilize the `ChannelConfirmed` event downstream in `LdkLite`.

**Comment by godsmith99x** [comment]
No, I got a new job a couple of months back and I don’t have the bandwidth.

**Comment by tnull** [comment]
> No, I got a new job a couple of months back and I don’t have the bandwidth. Sent from Proton Mail for iOS

Alright, no problem!
